mixnet/client/src/sender.rs

```diff
@@ -109,13 +130,83 @@ impl<R: Rng> Sender<R> {
 
         let mu: std::sync::Arc<tokio::sync::Mutex<tokio::net::TcpStream>> =
             pool.get_or_init(&addr).await?;
+        let arc_socket = mu.clone();
         let mut socket = mu.lock().await;
-        let body = Body::new_sphinx(packet);
-        body.write(&mut *socket).await?;
+        let body = Body::SphinxPacket(packet);
+        match body.write(&mut *socket).await {
+            Ok(_) => {
+                tracing::info!("Sent a Sphinx packet successuflly to the node: {addr}");
+                Ok(())
+            }
+            Err(e) => {
+                if let Some(err) = e.downcast_ref::<std::io::Error>() {
+                    match err.kind() {
+                        ErrorKind::BrokenPipe
+                        | ErrorKind::NotConnected
+                        | ErrorKind::ConnectionAborted => {
```

Comment:
`ErrorKind` variants grow continuously, not sure it's a good idea to whitelist them (e.g. we're missing `ErrorKind::Interrupted`". Maybe let's do the opposite and blacklist errors for which we don't want to retry (e.g. `ErrorKind::Unsupported`?

Reply:
Yeah, the match is messy here.